src/RA_StringUtils.h

```diff
@@ -259,18 +280,23 @@ class StringBuilder
 
         if constexpr (std::is_integral_v<T>)
         {
-            if (sFormat.front() == '0')
-                oss << std::setfill('0');
-            int nDigits = std::stoi(sFormat.c_str());
-            if (nDigits > 0)
-                oss << std::setw(nDigits);
-
-            if (sFormat.back() == 'X')
-                oss << std::uppercase << std::hex;
-            else if (sFormat.back() == 'x')
-                oss << std::hex;
-
-            oss << arg;
+            if constexpr (!is_char_v<T>)
```

Comment:
This logic should be confined to "%c" format. I'd expect these tests to be added to `TestStringPrintf` and `TestWStringPrintf`:
```
        Assert::AreEqual(std::string("abc"), StringPrintf("%c%c%c", 'a', 'b', 'c'));
        Assert::AreEqual(std::string("616263"), StringPrintf("%02x%02x%02x", 'a', 'b', 'c'));
```
```
        Assert::AreEqual(std::wstring(L"abc"), StringPrintf(L"%c%c%c", 'a', 'b', 'c'));
        Assert::AreEqual(std::wstring(L"616263"), StringPrintf(L"%02x%02x%02x", 'a', 'b', 'c'));
```

Reply:
Took me awhile to figure it out but I got it to work